openmc/surface.py

```diff
@@ -335,29 +412,45 @@ def z0(self, z0):
         self._coeffs['z0'] = z0
 
 
-
 class Cylinder(Surface):
+    """A cylinder whose length is parallel to the x-, y-, or z-axis.
```

Comment:
I've never actually used abstract base classes in Python, but I noticed something when I started playing around with them. Namely, adding this declaration doesn't actually prevent you from instantiating the class. You specifically have to add the `@abstractmethod` decorator to a method in order for it to raise an exception when you try to call it from the base class. Not sure how we should approach it for the classes here that shouldn't be instantiated...